services/order.py

```diff
@@ -0,0 +1,34 @@
+from django.contrib.auth import get_user_model
+from django.db.models import QuerySet
+from db.models import User
+from django.db import transaction
+from db.models import Order, Ticket
+
+
+def create_order(
+        tickets: list[dict],
+        username: User,
+        date: str = None
+) -> None:
+    with transaction.atomic():
```

Comment:
Use transaction.atomic as decorator instead of decorator. It makes your code more readable